tests/test_record_resource.py

```diff
@@ -150,3 +183,61 @@ def test_read_deleted_record(client, input_record):
     response = client.get("/records/{}".format(recid), headers=HEADERS)
     assert response.status_code == 410
     assert response.json['message'] == "The record has been deleted."
+
+
+def test_read_record_with_non_existing_pid(client, input_record):
+    """Test read a record with a non existing pid."""
+
+    response = client.get("/records/randomid", headers=HEADERS)
+    assert response.status_code == 404
+
+    assert response.json["status"] == 404
+    assert response.json['message'] == "The pid does not exist."
+
+
+def test_read_record_with_unregistered_pid(app_with_custom_minter,
+                                           input_record):
+    """Test read a record with an unregistered pid."""
+
+    client = app_with_custom_minter.test_client()
+    # Create dummy record with unregistered pid value
+    response = client.post(
+        "/records", headers=HEADERS, data=json.dumps(input_record)
+    )
+    assert response.status_code == 201
+    recid = response.json["pid"]
+
+    response = client.get("/records/{}".format(recid), headers=HEADERS)
+    assert response.status_code == 404
+
+    assert response.json["status"] == 404
+    assert response.json['message'] == "The pid is not registered."
+
+
+def test_read_record_with_redirected_pid(client, input_record):
+    """Test read a record with a redirected pid."""
+    from invenio_pidstore.models import PersistentIdentifier
```

Comment:
Move to top-level

Nested imports are scary :scream_cat: :
- either they exist *because* there are circular dependencies in the codebase which is really not good
- or they *cause* circular dependencies because developers are then not aware of the circular imports

It's a small thing that helps so much in the long run :)
(there are also performance implications, but performance is way less important than the above)